fhir-bulkimportexport/src/main/java/com/ibm/watson/health/fhir/bulkimport/BulkImportBatchLet.java

```diff
@@ -110,173 +88,90 @@
 
-    /**
-     * @param cosClient
-     */
-    public void listBuckets() {
-        if (cosClient == null)
-            return;
-
-        log("listBuckets", "Buckets:");
-        final List<Bucket> bucketList = cosClient.listBuckets();
-
-        for (final Bucket bucket : bucketList) {
-            log("listBuckets", bucket.getName());
-        }
-    }
-
-    private String getItem(String bucketName, String itemName) {
-        S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
-
-        try (InputStreamReader in = new InputStreamReader(item.getObjectContent())) {
-            final int bufferSize = 1024;
-            final char[] buffer = new char[bufferSize];
-            final StringBuilder out = new StringBuilder();
-
-            for (;;) {
-                int rsz = in.read(buffer, 0, buffer.length);
-                if (rsz < 0)
-                    break;
-                out.append(buffer, 0, rsz);
-            }
-            logger.log(Level.FINER, out.toString());
-            return out.toString();
-        } catch (IOException ioe) {
-            log("getItem", "Error reading file " + itemName);
-            return null;
-        }
-    }
-
     /**
      * Main entry point.
      */
     @Override
     public String process() throws Exception {
-        String exitStatus = "Import Stopped before it's started!";
-        log("process", "Begin get CosClient!");
-        getCosClient();
+        String exitStatus;
 
+        cosClient = COSUtils.getCosClient(cosCredentialIbm, cosApiKeyProperty, cosSrvinstId, cosEndpintUrl,
+                cosLocation);
         if (cosClient == null) {
-            log("process", "Failed to get CosClient!");
+            logger.warning("process: Failed to get CosClient!");
             return "Failed to get CosClient!";
         } else {
-            log("process", "Succeed get CosClient!");
+            logger.finer("process: Got CosClient successfully!");
         }
 
         if (cosBucketName == null) {
             cosBucketName = "fhir-bulkImExport-Connectathon";
         }
-
         cosBucketName = cosBucketName.toLowerCase();
-
-        listBuckets();
-
         if (!cosClient.doesBucketExist(cosBucketName)) {
-            log("process", "Bucket not found!");
+            logger.warning("process: Bucket '" + cosBucketName + "' not found!");
+            COSUtils.listBuckets(cosClient);
             return "Bucket not found!";
         }
-
         if (fhirTenant == null) {
             fhirTenant = "default";
-            log("process", "Set tenant to default!");
+            logger.info("process: Set tenant to default!");
         }
-        
         if (fhirDatastoreId == null) {
             fhirDatastoreId = Constants.DEFAULT_FHIR_TENANT;
-            log("readItem", "Set DatastoreId to default!");
+            logger.info("process: Set DatastoreId to default!");
         }
-        
+
         FHIRConfiguration.setConfigHome("./");
```

Comment:
what's this set for? is this because it is running outside of 'fhir-server'?

Reply:
ha, good catch, this can be removed per discussion with John

Reply:
done!